### PR TITLE
add test, function for pet show page, applications index

### PR DESCRIPTION
**Pull request by elguapogordo**
added test for pet show page to have link to associated applications index
  * using multiple applications - applications were made using models/objects instead of clicking
  * add link to show page
  * application index page lists pet_application.name as a link to that pet_application show page
added function for this test.

### CODE REVIEW

File: app/controllers/pet_show_applications_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class PetShowApplicationsController < ApplicationController
+
+  def index
+    @pet = Pet.find(params[:id])
+  end
```
Comment (elguapogordo) [inline app/controllers/pet_show_applications_controller.rb]: just threw in the pet object - we can link to everything else through the pet

File: config/routes.rb [hunk 2]
```diff
@@ -34,4 +34,6 @@
   patch '/favorites/:pet_id', to: 'favorites#update'
   delete '/favorites', to: 'favorites#destroy'
 
+  get '/pets/:id/applications', to: 'pet_show_applications#index'
```
Comment (elguapogordo) [inline config/routes.rb]: this really should have been pet_applications controller, but we already used that name. pet_applications really should have been named user_applications.  just a minor annoyance that falls in the "refactor later if we have time" category...

File: spec/features/applications/application_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      "city"=>"Jesse", "state"=>"CO", "zip"=>"80120", "phone_number"=>"303-867-5309",
+      "description"=>"Because I'm too cool for school"})
+
+    application_2 = PetApplication.create({"name"=>"Mike", "address"=>"12345 Mike Ave",
+      "city"=>"Mikeville", "state"=>"CO", "zip"=>"80120", "phone_number"=>"123-456-7890",
+      "description"=>"They're my pets already."})
+
+    ApplicationPet.create(pet_id: pet_1.id, pet_application_id: application_1.id)
+    ApplicationPet.create(pet_id: pet_1.id, pet_application_id: application_2.id)
+    ApplicationPet.create(pet_id: pet_2.id, pet_application_id: application_2.id)
```
Comment (elguapogordo) [inline spec/features/applications/application_index_spec.rb]: ha! two applications for three pets in ten lines! that wasn't going to happen by directing the test to click and fill in fields. also the fact that we wouldn't have been able to apply for the same pet twice that way. i like how this turned out.

File: config/routes.rb (hunk 2 above)
Comment (mikez321) [reply config/routes.rb]: looking like we might not have time, but commit the lesson to memory.... like not naming anything "application". :)

File: spec/features/applications/application_index_spec.rb (hunk 3 above)
Comment (mikez321) [reply spec/features/applications/application_index_spec.rb]: Really excited that you figured out how to make that work.  Good thinking for sure!

File: config/routes.rb (hunk 2 above)
Comment (elguapogordo) [reply config/routes.rb]: by the way, this line is the merge "conflict." there were no deletions, _just_ one line added. there _is_ no conflict.

**Pull request merged**